code/nick/python/lab13.py

```diff
@@ -0,0 +1,118 @@
+class Compass:
+    def __init__(self, heading) -> None:
+        if heading >= 360:
+            heading -= 360
+        self.heading = heading
+
+    def get_direction(self) -> str:
+        '''Find the nearest cardinal direction of the current heading'''
+        direction = round(self.heading/45)*45
+        if direction == 360:
+            direction = 0
+
+        match direction:
+            case 0:
+                return 'N'
+            case 45:
+                return 'NE'
+            case 90:
+                return 'E'
+            case 135:
+                return 'SE'
+            case 180:
+                return 'S'
+            case 225:
+                return 'SW'
+            case 270:
+                return 'W'
+            case 315:
+                return 'NW'
+
+    def turn(self, degrees, direction) -> None:
+        match direction:
+            case 'left':
+                self.heading -= degrees
+                if self.heading < 0:
+                    self.heading += 360
+
+            case 'right':
+                self.heading += degrees
+                if self.heading > 360:
+                    self.heading -= 360
+
+        print(
+            f'Turned {degrees} degrees {direction}. New heading is {self.heading}, pointed roughly {self.get_direction()}')
+
+    def __add__(self, degrees: int) -> int:
+        '''Add two different compass headings together'''
+        sum = self.heading + Compass(degrees).heading
```

Comment:
this seems unnecessarily complex, you're doing something much simpler for subtraction that is achieving the same result